### PR TITLE
Added: Like,history and watchlater page with all three functionalities

### PR DESCRIPTION
**Pull request by Dev-AkshatGupta**
src/pages/watchlater
src/pages/like
src/pages/singleVideoPlayer
src/pages/history
src/utilities/customHooks
These files needed to be reviewed
Thanks for reviewing my pr

### CODE REVIEW

**Review by aayushjain-code — APPROVED** [review]
Overall Code looks Clean You can..
UI of cards looks Impressive Keep going bro🚀🚀

File: src/contextAndReducers/DataProvider.jsx [hunk 1]
```diff
@@ -30,7 +30,6 @@ const DataProvider = ({ children }) => {
           const modifiedData = state.backUpData.filter(
             (item) => item.categoryName === action.payload
```
Comment (Maianki) [inline src/contextAndReducers/DataProvider.jsx:31]: You can directly destructure category name here. For e.g ```({categoryName})=>categoryName===action.payload```

**Review by Maianki — APPROVED** [review]
Small PR nothing much to suggest just added one suggestion. You can go ahead with the merge.

File: src/contextAndReducers/DataProvider.jsx (hunk 1 above)
Comment (Dev-AkshatGupta) [reply src/contextAndReducers/DataProvider.jsx:31]: Yeah I know that purposely I didn't do it. Thanks for reviewing though

**Pull request merged**